### PR TITLE
Port cpustates processor + bits

### PR DESCRIPTION
**Pull request by setrofim**
- Port cpustates processor from WA2, fixing the big.LITTLE idle state mismatch
  issue.
- process command improvements
- job logging improvements
- trace-cmd parser refactor

### CODE REVIEW

File: wa/utils/cpustates.py [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+            delta = timestamp - self.last_timestamp
+            for cpu, (idle, freq) in enumerate(self.previous_states):
+                if idle == -1 and freq is not None:
+                    state = '{:07}KHz'.format(freq)
+                elif freq:
+                    state = '{}-{:07}KHz'.format(self.idle_state_names[cpu][idle], freq)
+                elif idle not in (None, -1):
+                    state = self.idle_state_names[cpu][idle]
+                else:
+                    state = 'unkown'
```
Comment (marcbonnici) [inline wa/utils/cpustates.py]: /s/ "unkown"->"unknown"

File: wa/utils/cpustates.py [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+                 These are typically reported as part of ``TargetInfo`` in
+                 WA output.
+    :param output_basedir: Base location for the output. This directory must
+                        exist and must not contain a directory of file
+                        named ``"power-states"``.
+    :param use_rations: By default, stats will be reported as percentages. Set
+                        this to ``True`` to report stats as decimals in the
+                        ``0 <= value <= 1`` instead.
+    :param no_idle: ``False`` if cpuidle and at least one idle state per CPU are
+                    enabled, should be ``False`` otherwise. This influencies the
```
Comment (marcbonnici) [inline wa/utils/cpustates.py]: ``False`` -> ``True``?

/s/ "influencies" -> "influences"

File: wa/utils/cpustates.py [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+                    assumptions about CPU's initial states. If not explicitly
+                    set, the value for thie will be guessed based on whether
```
Comment (marcbonnici) [inline wa/utils/cpustates.py]: /s/ "thie" -> "this"

File: wa/utils/cpustates.py [hunk 4]
```diff
@@ -0,0 +1,723 @@
...
+    output_directory = os.path.join(output_basedir, 'power-states')
+    if not os.path.isdir(output_directory):
+        os.mkdir(output_directory)
+
+    freq_dependent_idle_states = []
+    if split_wfi_states:
+        freq_dependent_idle_states = [0]
+
+    # init trace, processor, and reporters
+    # note: filter_markers is Flase here, even though we *will* filter by them. The
```
Comment (marcbonnici) [inline wa/utils/cpustates.py]: /s/ "Flase" -> "False"

File: wa/utils/cpustates.py [hunk 5]
```diff
@@ -0,0 +1,723 @@
...
+    def update(self, timestamp, core_states):  # NOQA
+        row = [timestamp]
+        for core, [idle_state, frequency] in enumerate(core_states):
+            if idle_state == -1:
+                if frequency == UNKNOWN_FREQUENCY or frequency is None:
+                    frequency = 0
+            elif idle_state is None:
+                frequency = 0
+            else:
+                frequency = 0
```
Comment (marcbonnici) [inline wa/utils/cpustates.py]: Is the `elif` required if `frequency` is going to be set to 0 anyway?

**Comment by setrofim** [comment]
Comments addressed.

**Pull request merged**